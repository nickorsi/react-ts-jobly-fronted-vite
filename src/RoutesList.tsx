import { Route, Routes } from "react-router-dom"
import Home from "./Home"
import { CompanyList } from "./CompanyList"
import { Company } from "./Company"
import { JobList } from "./JobList"
import NotFound from "./NotFound"

/**
 * RoutesList componenet relying on the react-dom-router dependency.
 *
 * Maps componenets to relevant paths on the client side to mimic the
 * URL behavior of traditional html rendering sites.
 *
 * App -> RoutesList -> {Home, CompanyList, JobList, NotFound}
 */
function RoutesList() {
    return (
        <Routes>
            <Route path='/' element={ <Home/> }></Route>
            <Route path='companies' element={ <CompanyList/> }></Route>
            <Route path='companies/:companyHandle' element={ <Company/> }></Route>
            <Route path='jobs' element={ <JobList/> }></Route>
            <Route path='*' element={ <NotFound/>}></Route>
        </Routes>
    )
}

export default RoutesList;