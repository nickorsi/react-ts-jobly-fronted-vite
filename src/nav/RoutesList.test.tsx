import {describe, expect, it} from "vitest";
import { render, screen, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import RoutesList from "./RoutesList.tsx";

/**
 * Something wrong with test config, uncomment line below to get intellisense to
 * recognize the extended expect methods, comment out for tests to run properly.
*/
// import "@testing-library/jest-dom";

describe("Simple Render Tests", () => {
    it("renders without crashing", function () {
        render(
            <MemoryRouter >
                <RoutesList ></RoutesList>
            </MemoryRouter>
        );
    });
    it("matches the snapshot", function () {
        const { container } = render(
            <MemoryRouter >
                <RoutesList ></RoutesList>
            </MemoryRouter>
        );

        expect(container).toMatchSnapshot();
    });
});

describe("Test Individual Routes", function() {
    it("renders the home page correctly", function() {
        render(
            <MemoryRouter >
                <RoutesList ></RoutesList>
            </MemoryRouter>
        );

        expect(
            screen.getByText("All the jobs in one convenient place.")
        ).toBeInTheDocument();
    });
    it("renders the companies page correctly", async function() {
        render(
            <MemoryRouter initialEntries={["/companies"]}>
                <RoutesList ></RoutesList>
            </MemoryRouter>
        );

        await waitFor(() => {
            screen.debug();
            expect(
                screen.getByText("All Companies")
            ).toBeInTheDocument();
        });
    });
    it("renders the company page correctly", function() {
        render(
            <MemoryRouter initialEntries={["/companies/c1"]}>
                <RoutesList ></RoutesList>
            </MemoryRouter>
        );

        expect(
            screen.getByText("All the jobs in one convenient place.")
        ).toBeInTheDocument();
    });
    it("renders the jobs page correctly", function() {
        render(
            <MemoryRouter >
                <RoutesList ></RoutesList>
            </MemoryRouter>
        );

        expect(
            screen.getByText("All the jobs in one convenient place.")
        ).toBeInTheDocument();
    });
    it("renders the 404 page correctly", function() {
        render(
            <MemoryRouter >
                <RoutesList ></RoutesList>
            </MemoryRouter>
        );

        expect(
            screen.getByText("All the jobs in one convenient place.")
        ).toBeInTheDocument();
    });
});