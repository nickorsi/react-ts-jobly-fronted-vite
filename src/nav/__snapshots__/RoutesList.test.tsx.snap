// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Simple Render Tests > matches the snapshot 1`] = `
<div>
  <div
    class="home card"
  >
    <h1>
      Jobly ðŸ’¼
    </h1>
    <p>
      All the jobs in one convenient place.
    </p>
    <p>
      So what are you waiting for? Find your next dream job here!
    </p>
  </div>
</div>
`;
